### DISCUSSION TITLE
Refreshing Cockpit plugin index.html without restart?

### DISCUSSION THREAD

**Discussion by centminmod** [cockpit-project/cockpit#19702] [Q&A] [answered]
Started playing with Cockpit custom plugin as outlined at https://cockpit-project.org/blog/creating-plugins-for-the-cockpit-user-interface.html. But if I create `/usr/share/cockpit/pluginname/index.html` and update that index.html source code, is it possible to have refreshed updated index.html displayed on Cockpit dashboard page without running `systemctl try-restart cockpit`?

**Comment by martinpitt** [comment, marked as answer]
You don't have to restart the service at all. In the worst case, log out and back in. However, it is enough to just force-refresh the page with (Ctrl+Shift+R in Firefox and Chromium), which provides much faster iteration.

Also, please don't copy your own custom pages to /usr/share/, that's package manager territory. If you keep building and installing an rpm that's ok, but it's much faster to just symlink the build tree into ~/.local/share/cockpit, with `make devel-install`, see https://github.com/cockpit-project/starter-kit/blob/main/README.md#installing

**Reply by centminmod** [reply]
Thanks @martinpitt for the reply and info. Much appreciated 👍
